webrtc/pending-promise-after-close.html

```diff
@@ -0,0 +1,81 @@
+<!doctype html>
+<title>Test pending promises behavior when RTCPeerConnection object is closed</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script type="text/javascript">
+  'use strict';
+
+  function test_never_resolved(test_func, testName) {
+    async_test(t => {
+      const pc1 = new RTCPeerConnection();
+      const closePc1 = () => pc1.close();
+
+      test_func(pc1, closePc1)
+      .then(
+        t.step_func(result => {
+          assert_unreached(`Pending promise should never be resolved. Instead it is fulfilled with: ${result}`);
+        }),
+        t.step_func(err => {
+         assert_unreached(`Pending promise should never be resolved. Instead it is rejected with: ${err}`);
+        }));
+
+      t.step_timeout(() => {
+        t.done();
+      }, 1000);
+
+      const pc2 = new RTCPeerConnection();
+      const closePc2 = () => {}; // noop (don't close)
+
+      test_func(pc2, closePc2)
+      .then(
+        t.step_func_done(),
+        t.step_func(err => {
+          assert_unreached(`Operating on unclosed peer connection should succeed. Instead it is rejected with: ${err}`);
+        }));
```

Comment:
Oh, I did this after @foolip's earlier comment that the timeout for the unreached case can be eliminated by waiting for the success case promise. So if you notice both line 23 and 31 terminate the test successfully. I can add in some comments and modify the title to make it more clear.

> One way to avoid timeouts in a test like this is to use another chain of events that we know would be completed after the first. So, you could create a second RTCPeerConnection where you run the same steps and then don't call close(). Once that is done, if the first hasn't been resolved or rejected yet, then the test can end. Probably the test would run faster, although use more resources. Whatever you think would be nicer.